### ISSUE TITLE
Implement arcade-powered source-build stage 2: CI

### ISSUE THREAD

**Issue by dagood** [dotnet/arcade#6845]
- [ ] This issue is blocking 
- [ ] This issue is causing unreasonable pain

Blocks arcade-powered source-build: https://github.com/dotnet/source-build/blob/master/Documentation/planning/arcade-powered-source-build/implementation-plan.md

---

Stage 2 is adding source-build CI to `master`: https://github.com/dotnet/source-build/blob/master/Documentation/planning/arcade-powered-source-build/onboarding/ci-onboarding.md

In the ideal case, this is a one-line add to the place where the repo uses the the arcade CI template. This is pretty likely for dotnet/arcade since AFAIK it should be dogfooding the ordinary template. I recommend a mock official build to make sure the source-build CI will work there too.

More general info about onboarding and the plan starts here: https://github.com/dotnet/source-build/tree/master/Documentation/planning/arcade-powered-source-build/onboarding

**Comment by MattGal** [comment]
I agree this is a small change but I'm unsure how it fits into FR's charter.

**Comment by dagood** [comment]
After a bit more discussion (@ilyas1974, source-build folks), we don't think this fits the FR model. This particular issue is small enough that FR could do it, but source-build is looking to establish someone who can help later on with arcade-powered source-build:

* Removing `.patch`es. (This does not block, but it is critical to dotnet/arcade maintainability.) (https://github.com/dotnet/arcade/pull/6841)
* Reviewing + merging source-build infra PRs. (Ideally this lets the person get familiar with this infra so they can help with repo-specific issues e.g. build failures in the future. But source-build folks won't disappear.)
* Completing (or driving completion) for any other "repo owner" labeled parts of the remaining arcade-powered source-build stages.

This is specific to dotnet/arcade. (Ideally this would extend to dotnet/xliff-tasks, which has the same ownership as dotnet/arcade, but finding a second person for xliff-tasks would be fine.)

We're still able to get other parts of arcade-powered source-build done while waiting on this issue, so I've unmarked "blocking". This issue blocks completion of the overall effort but it's not making any devs sit around wasting time (yet 😄). We think it's worth waiting to get the right help--if FR implements this, we've only kicked the problem a little down the road.

(For super-broad context, this is for this P0 6.0 user story: https://github.com/dotnet/core/issues/5448.)
